### PR TITLE
Sb annotation tools

### PR DESCRIPTION
**Pull request by spencerAJ**
Pivotal Tracker: 
[#176099742 - Create basic layout for the primary toolbar](https://www.pivotaltracker.com/story/show/176099742),
[#176100795 - Create annotation tools (Selector, Point Comment, Highlight, Text, Strikeout, Pen, Rectangle)](https://www.pivotaltracker.com/story/show/176100795),
[#176099932 - Create the secondary toolbar with Color Picker, Size Picker, and Line Picker](https://www.pivotaltracker.com/story/show/176099932),
[#176100719 - Create document control tools (rotate, zoom, full screen)](https://www.pivotaltracker.com/story/show/176100719)

Adds most the front-end structure and functionality for the toolbars and the annotation/document control tools it contains.

### CODE REVIEW

File: client/apps/docviewer/components/main/docviewer.jsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          RENDER_OPTIONS={renderOptions}
+          handleRerender={this.handleRerender}
+          handleFullScreen={this.handleFullScreen}
+        />
+        <div className="toolbar" />
+        <input
+          type="file"
+          onChange={this.handleFileInput}
+        />
+        <div id="viewer" className="pdfViewer" />
```
[the remark this answers is not in the document]
Comment (spencerAJ) [reply client/apps/docviewer/components/main/docviewer.jsx:112]: I think these warnings are from the react fragment as they should be double quote

File: db/seeds.rb [hunk 2]
```diff
@@ -448,6 +448,55 @@
...
+      title: "Docviewer",
+      description: "Docviewer Application",
+      privacy_level: "public",
+      icon: "oauth_icon.png",
+      custom_fields: {
+        canvas_account_id: "$Canvas.account.id",
+        external_tool_url: "$Canvas.externalTool.url",
+        canvas_user_id: "$Canvas.user.id",
+      },
+      course_navigation: {
```
Comment (srphillips) [inline db/seeds.rb:487]: Let's ask about where we can make placement work here, so we don't have to wipe out seeds later.

File: package.json [hunk 3]
```diff
@@ -13,7 +13,8 @@
     "lint": "eslint client",
     "lint_fix": "eslint client --fix",
     "nuke": "rm -rf node_modules",
-    "inspect_fuel": "\"${EDITOR:-vi}\" ./node_modules/atomic-fuel"
+    "inspect_fuel": "\"${EDITOR:-vi}\" ./node_modules/atomic-fuel",
+    "pdf": "node utils/cp.js node_modules/pdfjs-dist/build/pdf.worker.min.js client/apps/docviewer/pdf.worker.min.data"
```
Comment (srphillips) [inline package.json:17]: Is this documented anywhere as necessary in the build process?

**Review by srphillips — COMMENTED** [review]
The way that Github shows diffs of the migrations is a little odd.  Traditionally we wouldn't edit migrations, but this is pre-release and it looks like the new migrations replaced previous old timestamped ones.

**Comment by srphillips** [comment]
I'm not concerned about those particular hound errors, cause they seem wrong.  I am concerned that the Travis build failed and would like that to succeed before a merge.

File: package.json (hunk 3 above)
Comment (spencerAJ) [reply package.json:17]: I think I was trying to add a script that the Submitty fork was using to copy over worker files but it ended up not being needed. I'll remove it.

File: db/seeds.rb (hunk 2 above)
Comment (spencerAJ) [reply db/seeds.rb]: I must have had something set up differently in the global JS before. I removed the course navigation placement and it seems to be working fine now.

File: package.json [hunk 4]
```diff
@@ -93,7 +91,7 @@
     "mime": "^2.4.4",
     "moment": "^2.22.2",
     "moment-timezone": "^0.5.27",
-    "pdf-annotate.js": "file:../pdf-annotate.js",
+    "pdf-annotate.js": "atomicjolt/pdf-annotate.js#sb-tool-fixes",
```
Comment (srphillips) [inline package.json:94]: I'd like to get this changed to a stable branch/tag before merging.

**Pull request merged**